multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -138,10 +142,41 @@ func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctr
 				}
 			}
 		}
-		// If err != nil, probably ClusterClaims were deleted during the processing of MemberClusterAnnounce.
-		// Nothing to handle in this case and MemberClusterAnnounce will also be deleted soon.
-		// TODO: Add ClusterClaim webhook to make sure it cannot be deleted while ClusterSet is present.
 	}
+	// If err != nil, probably ClusterClaims were deleted during the processing of MemberClusterAnnounce.
+	// Nothing to handle in this case and MemberClusterAnnounce will also be deleted soon.
+	// TODO: Add ClusterClaim webhook to make sure it cannot be deleted while ClusterSet is present.
+
+	finalizer := fmt.Sprintf("%s/%s", MemberClusterAnnounceFinalier, memberAnnounce.ClusterID)
+	if memberAnnounce.DeletionTimestamp.IsZero() {
+		if controllerutil.ContainsFinalizer(memberAnnounce, finalizer) {
```

Comment:
We already have a method `StringExistsInSlice` in common package, you can use that instead of introducing a new dependency.

Reply:
Remove pkg controllerutils. Using `common.StringExistsInSlice` and `common.RemoveStringFromSlice` intead of the pkg.